### ISSUE TITLE
R linter run by super-linter gives false positives

### ISSUE THREAD

**Issue by kpagacz** [github/super-linter#1910]
## Describe the bug

super-linter gives false positives when linting R language. super-linter uses [lintr](https://github.com/jimhester/lintr/) to lint R language files. If a linted github repository contains an R library (package), then lintr assumes that the package is installed. Otherwise, it detects false positives (specifically lintr's object_usage_linter).

## Expected behavior

super-linter does not return false positives when linting R language.

## Steps to Reproduce

I have run super-linter docker locally with the recommended arguments to provide a log. Here is the repo with the log: https://github.com/kpagacz/superlintertest

This particular lint is not present when lintr is run locally after installing the package (excerpt from super-linter.log)
```
2021-08-31 04:36:28 [ERROR]   Found errors in [R] linter!
2021-08-31 04:36:28 [ERROR]   Error code: 31. Command output:
------
/tmp/lint/R/test.R:5:9: warning: no visible binding for global variable ‘NS’
  ns <- NS
        ^~
------
2021-08-31 04:36:28 [DEBUG]   Error code: 31. Command output:
------
/tmp/lint/R/test.R:5:9: warning: no visible binding for global variable ‘NS’
  ns <- NS
        ^~
------
```

Excerpt from my command line:
```
r$> devtools::install('superlintertest')            
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                                
2: CRAN packages only                 
3: None                               
4: R6        (2.5.0   -> 2.5.1) [CRAN]
5: htmltools (0.5.1.1 -> 0.5.2) [CRAN]
6: later     (1.2.0   -> 1.3.0) [CRAN]
7: cachem    (1.0.5   -> 1.0.6) [CRAN]
8: httpuv    (1.6.1   -> 1.6.2) [CRAN]

Enter one or more numbers, or an empty line to skip 
updates: 3                                          
   checking for file ‘/home/rstudio/projects/superli✔  checking for file ‘/home/rstudio/projects/superlintertest/DESCRIPTION’
─  preparing ‘superlintertest’: (366ms)
✔  checking DESCRIPTION meta-information
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
   Omitted ‘LazyData’ from DESCRIPTION
─  building ‘superlintertest_0.0.0.9000.tar.gz’
   
Running /usr/local/lib/R/bin/R CMD INSTALL \
  /tmp/Rtmp8AWMHc/superlintertest_0.0.0.9000.tar.gz \
  --install-tests 
* installing to library ‘/usr/local/lib/R/site-library’
* installing *source* package ‘superlintertest’ ...
** using staged installation
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (superlintertest)

r$> lintr::lint_package('superlintertest')
.          
```

## Logs

Logs are in the linked repository, but here is the link to the file anyway: https://github.com/kpagacz/superlintertest/blob/main/super-linter.log

## Additional context

We are using super-linter to lint our R packages in the organization and we would really appreciate if super-linter behaved the same way as do the native R linting tools.

**Comment by kpagacz** [comment]
R provides command line tools to install an R package. They are respectively R CMD BUILD <path> to build a library in the <path> and R CMD INSTALL <path_to_built_package> to install the built package.
